source/Interfaces/NetWorth/client/ClientInterface.hpp

```diff
@@ -0,0 +1,161 @@
+/**
+ * This file is part of the Fall 2023, CSE 491 course project.
+ * @brief A networking interface that allows information to be sent across a  network
+ * @note Status: PROTOTYPE
+ **/
+
+#pragma once
+
+#include <cassert>
+#include <string>
+#include <vector>
+#include <SFML/Network/UdpSocket.hpp>
+#include <SFML/Network/Packet.hpp>
+#include <memory>
+
+#include "../NetworkInterface.hpp"
+#include "../../TrashInterface.hpp"
+
+using namespace sf;
+
+namespace cse491 {
+    namespace netWorth{
+    /**
+     * The interface of our client that will be interacting and connection with our server
+     */
+        class ClientInterface {
+        private:
+            std::unique_ptr<TrashInterface> mTrash;     /// Display interface
+
+            sf::IpAddress mIp;      /// Destination IP address
+            sf::UdpSocket mSocket;  /// UDP socket for sending and receiving
+            unsigned short mPort;   /// Destination port
+
+        protected:
+
+
+        public:
+            /**
+             * ClientInterface constructor (NetworkingInterface superclass)
+             * @param ip_string String for destination IP address, make into IpAddress object
+             * @param port Destination port number
+             */
+            ClientInterface(const std::string & ip_string,
+                            unsigned short port) {
+                mIp = sf::IpAddress(ip_string);
+                mPort = port;
+            }
+
+            /**
+             * Default destructor
+             */
+            ~ClientInterface() = default;
+
+            /**
+             * Establish connection with server
+             * @return True if successful, false if error
+             */
+            bool EstablishConnection() {
+                // send ping to server
+                sf::Packet send_pkt;
+                sf::Packet recv_pkt;
+                std::string str = "Ping!";
+                send_pkt << str;
+
+                if (mSocket.send(send_pkt, mIp, mPort) != Socket::Status::Done) {
+                    std::cout << "Could not connect to " << mIp << " at port " << mPort << std::endl;
+                    return false;
+                }
+
+                // receive pong from server
+                if (mSocket.receive(recv_pkt, mIp, mPort) != sf::Socket::Done)
+                {
+                    std::cout << "Failure to receive" << std::endl;
+                    return false;
+                }
+                recv_pkt >> str;
+                std::cout << str << std::endl;
+
+                return true;
+            }
+
+            /**
+             * Game loop
+             */
+            void Loop() {
```

Comment:
I'd push for a more descriptive name for this function than just `Loop()` since that's as generic as it comes.  At least call it `GameLoop()` which would provide some more information.

This function is also quite long; I'd recommend building a set of helper functions to make it more readable.  For example, you can have `GameLoop_Setup()`, `GameLoop_GetInput()`, `GameLoop_ProcessInput()`, `GameLoop_SendAction()`, etc.  I'm happy to provide more detailed suggestions if that would be helpful.

Reply:
Ended up making a GameLoop_GetInput() function, that was really helpful.

Also added some helper functions in the NetworkingInterface class. Every time I would send or receive a packet, a status code is returned and I would check it and print an error if something was wrong. I moved the check and error message into one function and made it return a bool to check for success (I still have to quit the loop if there is an error). Having the if statements for each send/receive looked pretty bulky. Can probably do more helper functions later.